src/main/java/gyro/google/compute/SecurityPolicyRuleRateLimitOptionsConfig.java

```diff
@@ -0,0 +1,302 @@
+import gyro.google.Copyable;
+
+public class SecurityPolicyRuleRateLimitOptionsConfig extends Diffable
+    implements Copyable<SecurityPolicyRuleRateLimitOptions> {
+
+    private SecurityPolicyRuleRateLimitOptionsThresholdConfig banThreshold;
+    private Integer banDurationSec;
+    private SecurityPolicyRuleRateLimitOptionsThresholdConfig rateLimitThreshold;
+    private String conformAction;
+    private String exceedAction;
+
+    // Not supported by the api yet
+    private String enforceOnKey;
+    private String enforceOnKeyName;
+
+    // Not supported in the UI yet
+    private SecurityPolicyRuleRedirectOptionsConfig exceedRedirectConfig;
+
+    /**
+     * The ban threshold config. Only valid when rule ``action`` is set to ``rate_based_ban``.
+     *
+     * @subresource gyro.google.compute.SecurityPolicyRuleRateLimitOptionsThresholdConfig
+     */
+    @Updatable
+    public SecurityPolicyRuleRateLimitOptionsThresholdConfig getBanThreshold() {
+        return banThreshold;
+    }
+
+    public void setBanThreshold(SecurityPolicyRuleRateLimitOptionsThresholdConfig banThreshold) {
+        this.banThreshold = banThreshold;
+    }
+
+    /**
+     * The ban duration in seconds. Only valid when rule ``action`` is set to ``rate_based_ban``.
+     */
+    @Updatable
+    public Integer getBanDurationSec() {
+        return banDurationSec;
+    }
+
+    public void setBanDurationSec(Integer banDurationSec) {
+        this.banDurationSec = banDurationSec;
+    }
+
+    /**
+     * The rate limit threshold config.
+     *
+     * @subresource gyro.google.compute.SecurityPolicyRuleRateLimitOptionsThresholdConfig
+     */
+    @Required
+    @Updatable
+    public SecurityPolicyRuleRateLimitOptionsThresholdConfig getRateLimitThreshold() {
+        return rateLimitThreshold;
+    }
+
+    public void setRateLimitThreshold(SecurityPolicyRuleRateLimitOptionsThresholdConfig rateLimitThreshold) {
+        this.rateLimitThreshold = rateLimitThreshold;
+    }
+
+    /**
+     * The action to take when requests are under the threshold. Default is ``allow``.
+     */
+    @Required
+    @ValidStrings("allow")
+    public String getConformAction() {
+        if (conformAction == null) {
+            conformAction = "allow";
+        }
+
+        return conformAction;
+    }
+
+    public void setConformAction(String conformAction) {
+        this.conformAction = conformAction;
+    }
+
+    /**
+     * The action to take when requests are over the threshold.
+     */
+    @Required
+    @Updatable
+    @ValidStrings({"deny(403)", "deny(404)", "deny(429)", "deny(502)", "redirect"})
+    public String getExceedAction() {
+        return exceedAction;
+    }
+
+    public void setExceedAction(String exceedAction) {
+        this.exceedAction = exceedAction;
+    }
+
+    /**
+     * The key to enforce the rate limit on.
+     */
+    @Updatable
+    public String getEnforceOnKey() {
+        return enforceOnKey;
+    }
```

Comment:
```suggestion
    /**
     * The key to enforce the rate limit on.
     */
    @Updatable
    @ValidStrings({ "ALL", "IP", "HTTP_HEADER", "XFF_IP", "HTTP_COOKIE", "HTTP_PATH", "SNI", "REGION_CODE" })
    public String getEnforceOnKey() {
        return enforceOnKey;
    }
```


Has the particular valid Strings:

Determines the key to enforce the rate_limit_threshold on. Possible values are:
- ALL: A single rate limit threshold is applied to all the requests matching this rule. This is the default value if "enforceOnKey" is not configured.
- IP: The source IP address of the request is the key. Each IP has this limit enforced separately.
- HTTP_HEADER: The value of the HTTP header whose name is configured under "enforceOnKeyName". The key value is truncated to the first 128 bytes of the header value. If no such header is present in the request, the key type defaults to ALL.
- XFF_IP: The first IP address (i.e. the originating client IP address) specified in the list of IPs under X-Forwarded-For HTTP header. If no such header is present or the value is not a valid IP, the key defaults to the source IP address of the request i.e. key type IP.
- HTTP_COOKIE: The value of the HTTP cookie whose name is configured under "enforceOnKeyName". The key value is truncated to the first 128 bytes of the cookie value. If no such cookie is present in the request, the key type defaults to ALL.
- HTTP_PATH: The URL path of the HTTP request. The key value is truncated to the first 128 bytes.
- SNI: Server name indication in the TLS session of the HTTPS request. The key value is truncated to the first 128 bytes. The key type defaults to ALL on a HTTP session.
- REGION_CODE: The country/region from which the request originates.
  Check the EnforceOnKey enum for the list of possible values.